backend/app/services/knowledge/task_knowledge_base_service.py

```
@@ -139,79 +142,73 @@ def _is_kb_bound_to_user_group_chat(
-        owned_tasks = (
-            db.query(TaskResource)
-            .filter(
+        # Subquery: tasks where user is approved member (join with TaskResource for active check)
+        member_task_subquery = (
+            select(ResourceMember.resource_id)
+            .join(TaskResource, TaskResource.id == ResourceMember.resource_id)
+            .where(
+                ResourceMember.user_id == user_id,
+                ResourceMember.resource_type == ResourceType.TASK.value,
+                ResourceMember.status == MemberStatus.APPROVED.value,
+                TaskResource.is_active == True,
```

Ruff (0.15.5)

[error] Avoid equality comparisons to `True`; use `TaskResource.is_active:` for truth checks

Replace with `TaskResource.is_active`

(E712)